Last-Update: 2013-12-15
Forwarded: not-needed
Author: Christoph Haas <haas@debian.org>
Reviewed-By: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: Make Zabbix put its generated config file into /etc/zabbix

Index: dotdeb-zabbix/frontends/php/include/classes/class.cconfigfile.php
===================================================================
--- dotdeb-zabbix.orig/frontends/php/include/classes/class.cconfigfile.php
+++ dotdeb-zabbix/frontends/php/include/classes/class.cconfigfile.php
@@ -24,7 +24,7 @@ class CConfigFile {
 	const CONFIG_NOT_FOUND = 1;
 	const CONFIG_ERROR = 2;
 
-	const CONFIG_FILE_PATH = '/conf/zabbix.conf.php';
+	const CONFIG_FILE_PATH = '/etc/zabbix/zabbix.conf.php';
 
 	public $configFile = null;
 	public $config = array();
Index: dotdeb-zabbix/frontends/php/include/setup.inc.php
===================================================================
--- dotdeb-zabbix.orig/frontends/php/include/setup.inc.php
+++ dotdeb-zabbix/frontends/php/include/setup.inc.php
@@ -443,7 +443,7 @@ class CSetupWizard extends CForm {
 	function stage6() {
 		$this->setConfig('ZBX_CONFIG_FILE_CORRECT', true);
 
-		$config = new CConfigFile(Z::getInstance()->getRootDir().CConfigFile::CONFIG_FILE_PATH);
+		$config = new CConfigFile(CConfigFile::CONFIG_FILE_PATH);
 		$config->config = array(
 			'DB' => array(
 				'TYPE' => $this->getConfig('DB_TYPE'),
@@ -510,7 +510,7 @@ class CSetupWizard extends CForm {
 		$this->HIDE_CANCEL_BUTTON = !$this->DISABLE_NEXT_BUTTON;
 
 
-		$table = array('Configuration file', BR(), '"'.Z::getInstance()->getRootDir().CConfigFile::CONFIG_FILE_PATH.'"',
+		$table = array('Configuration file', BR(), '"'.CConfigFile::CONFIG_FILE_PATH.'"',
 			BR(), 'created: ', $this->getConfig('ZBX_CONFIG_FILE_CORRECT', false)
 			? new CSpan(_('OK'), 'ok')
 			: new CSpan(_('Fail'), 'fail')
@@ -523,7 +523,7 @@ class CSetupWizard extends CForm {
 				? array($error_text, BR(), 'Please install it manually, or fix permissions on the conf directory.', BR(), BR(),
 					'Press the "Download configuration file" button, download the configuration file ',
 					'and save it as ', BR(),
-					'"'.Z::getInstance()->getRootDir().CConfigFile::CONFIG_FILE_PATH.'"', BR(), BR(),
+					'"'.CConfigFile::CONFIG_FILE_PATH.'"', BR(), BR(),
 					new CSubmit('save_config', 'Download configuration file'),
 					BR(), BR()
 				)
@@ -629,7 +629,7 @@ class CSetupWizard extends CForm {
 				// make zabbix.conf.php downloadable
 				header('Content-Type: application/x-httpd-php');
 				header('Content-Disposition: attachment; filename="'.basename(CConfigFile::CONFIG_FILE_PATH).'"');
-				$config = new CConfigFile(Z::getInstance()->getRootDir().CConfigFile::CONFIG_FILE_PATH);
+				$config = new CConfigFile(CConfigFile::CONFIG_FILE_PATH);
 				$config->config = array(
 					'DB' => array(
 						'TYPE' => $this->getConfig('DB_TYPE'),
Index: dotdeb-zabbix/frontends/php/include/classes/core/ZBase.php
===================================================================
--- dotdeb-zabbix.orig/frontends/php/include/classes/core/ZBase.php
+++ dotdeb-zabbix/frontends/php/include/classes/core/ZBase.php
@@ -289,7 +289,7 @@ class ZBase {
 	 * Load zabbix config file.
 	 */
 	protected function loadConfigFile() {
-		$configFile = $this->getRootDir().CConfigFile::CONFIG_FILE_PATH;
+		$configFile = CConfigFile::CONFIG_FILE_PATH;
 		$config = new CConfigFile($configFile);
 		$this->config = $config->load();
 	}
Index: dotdeb-zabbix/frontends/php/include/classes/items/CHelpItems.php
===================================================================
--- dotdeb-zabbix.orig/frontends/php/include/classes/items/CHelpItems.php
+++ dotdeb-zabbix/frontends/php/include/classes/items/CHelpItems.php
@@ -262,7 +262,7 @@ class CHelpItems {
 				),
 				array(
 					'key' => 'vfs.file.md5sum[file]',
-					'description' => _('Calculate MD5 check sum of a given file. String MD5 hash of the file. Can be used for files less than 64MB, unsupported otherwise. Example: vfs.file.md5sum[/usr/local/etc/zabbix_agentd.conf]')
+					'description' => _('Calculate MD5 check sum of a given file. String MD5 hash of the file. Can be used for files less than 64MB, unsupported otherwise. Example: vfs.file.md5sum[/etc/zabbix/zabbix_agentd.conf]')
 				),
 				array(
 					'key' => 'vfs.file.regexp[file,regexp,<encoding>,<start line>,<end line>,<output>]',
@@ -524,7 +524,7 @@ class CHelpItems {
 				),
 				array(
 					'key' => 'vfs.file.md5sum[file]',
-					'description' => _('Calculate MD5 check sum of a given file. String MD5 hash of the file. Can be used for files less than 64MB, unsupported otherwise. Example: vfs.file.md5sum[/usr/local/etc/zabbix_agentd.conf]')
+					'description' => _('Calculate MD5 check sum of a given file. String MD5 hash of the file. Can be used for files less than 64MB, unsupported otherwise. Example: vfs.file.md5sum[/etc/zabbix/zabbix_agentd.conf]')
 				),
 				array(
 					'key' => 'vfs.file.regexp[file,regexp,<encoding>,<start line>,<end line>,<output>]',
